loadable_apps/loadable_sample/wifiapp/recovery.c

```diff
@@ -44,18 +51,31 @@ static void *assert_thread(void *index)
 		PANIC();
 	} else if (type == 1) {
 		/* Access kernel code */
-		addr = (uint32_t *)CONFIG_MPU_TEST_KERNEL_CODE_ADDR;
-		printf("[%d] %dth thread, Write kernel code space 0x%x\n", getpid(), (int)index, addr);
+		obj.type = MPUTEST_KERNEL_CODE;
+		ret = ioctl(tc_fd, TESTIOC_MPUTEST, &obj);
+		if (ret < 0) {
+			printf("ERROR: Failed to obtain address from kernel\n");
+			return ERROR;
+		}
+
+		printf("[%d] %dth thread, Write kernel code space 0x%x\n", getpid(), (int)index, obj.addr);
 		sleep(1);
-		*addr = 0xdeadbeef;
+		*(obj.addr) = 0xdeadbeef;
 	} else {
 		/* Access another binary 'micom' address */
-		addr = (uint32_t *)CONFIG_MPU_TEST_APP_ADDR;
-		printf("[%d] %dth thread, Write another app space 0x%x\n", getpid(), (int)index, addr);
+		obj.type = MPUTEST_APP_ADDR;
+		ret = ioctl(tc_fd, TESTIOC_MPUTEST, &obj);
+		if (ret < 0) {
+			printf("ERROR: Failed to obtain address from kernel\n");
+			return ERROR;
+		}
+
+		printf("[%d] %dth thread, Write another app space 0x%x\n", getpid(), (int)index, obj.addr);
 		sleep(1);
-		*addr = 0xdeadbeef;
+		*(obj.addr) = 0xdeadbeef;
```

Comment:
here also you can reduce duplicated code.
and 0xdeadbeef is only hexspeak value which we use for color?